prime-router/src/main/kotlin/fhirengine/translation/hl7/utils/FhirBundleUtils.kt

```diff
@@ -40,7 +41,7 @@ object FhirBundleUtils : Logging {
      * Converts a [value] of type [sourceType] into a compatible Base of type [targetType]. Returns the original value
      * and logs an error if the conversion is not supported.
      */
-    fun convertFhirType(value: Base, sourceType: String, targetType: String): Base {
+    fun convertFhirType(value: Base, sourceType: String, targetType: String, logger: KotlinLogger = this.logger): Base {
```

Comment:
logger should not be needed as an argument. It is already provided by the class

Reply:
It's an optional argument, by default uses the one on the class, but accepts a different logger to assist in testing the logging with mocks. From your other comments it sounds like you're not a fan of testing the logging, so let's discuss this as a 17th and we can decide how to handle this.

Reply:
I do think this is needed here to help with the logging mocking, even if the specific messages still aren't being verified, it's nice to be able to verify that an error or warning is logged.